### PR TITLE
Document how to disallow blocking calls inside allowed blocking calls

### LINKED ISSUE #187: Provide example and better document disallowBlockingCallsInside
After some trial-and-error, here is an example that demonstrates the feature:

```java
public class BlockingDisallowTest {

    static {
        BlockHound.install(b -> b
                .allowBlockingCallsInside(NonBlockingClass.class.getName(), "outer")
                .disallowBlockingCallsInside(NonBlockingClass.class.getName(), "inner")
        );
    }

    @Test
    public void shouldDisallow() throws InterruptedException {
        NonBlockingClass nbc = new NonBlockingClass();
        AtomicReference<BlockingOperationError> boeRef = new AtomicReference<>();

        //to trip BlockHound in the first place, we must be in a nonblocking thread
        CountDownLatch latch = new CountDownLatch(1);
        Schedulers.parallel().schedule(() -> {
            try {
                nbc.outer();
            }
            catch (BlockingOperationError boe) {
                boeRef.set(boe);
            }
            finally {
                latch.countDown();
            }
        });

        latch.await(5, TimeUnit.SECONDS);

        //given the configuration we expect that yield is allowed, but the sleep inside example2 isn't
        assertThat(boeRef.get())
                .isNotNull()
                .hasMessage("Blocking call! java.lang.Thread.sleep")
                .hasStackTraceContaining("at com.example.BlockingDisallowTest$NonBlockingClass.inner")
                .hasStackTraceContaining("at com.example.BlockingDisallowTest$NonBlockingClass.outer");
    }

    static class NonBlockingClass {

        String inner() {
            try {
                //if this trips BlockHound, the test fails (inner not in the stacktrace)
                Thread.sleep(50);
            }
            catch (InterruptedException e) {
                e.printStackTrace();
            }
            return "example";
        }

        String outer() {
            try {
                Thread.sleep(50);
            }
            catch (InterruptedException e) {
                e.printStackTrace();
            }
            Thread.yield();
            return inner();
        }
    }
}
```

It should be noted that _disallow_ seems to only make sense as an `inner` exception to an `outer` _allow_ case:
 - the reverse can simply be achieved by NOT allowing blocking calls inside `outer` in the first place, and merely allowing `inner`
 - it doesn't mark the method as a non-blocking execution context. the method MUST still be called from a thread that is already detected as non-blocking by BlockHound (eg. via the `threadPredicate`) for the allow/disallow evaluation to take place

### PR DESCRIPTION
**Pull request by pderop**
This PR documents how to disallow blocking calls that are done from allowed blocking calls.
Also committed the junit test proposed by Simon (thanks).

Fixes #187

### CODE REVIEW

File: example/src/test/java/com/example/BlockingDisallowTest.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            } catch (BlockingOperationError boe) {
+                boeRef.set(boe);
+            } finally {
+                latch.countDown();
+            }
+        });
+
+        latch.await(5, TimeUnit.SECONDS);
+
+        //given the configuration we expect that yield is allowed, but the sleep inside example2 isn't
```
Comment (chemicL) [inline example/src/test/java/com/example/BlockingDisallowTest.java:58]:
```suggestion
        //given the configuration we expect that Thread.yield() is allowed, but Thread.sleep() inside inner() isn't.
```

**Review by chemicL — APPROVED** [review]
Minor comment improvement suggestion, otherwise lookin' good!

File: example/src/test/java/com/example/BlockingDisallowTest.java (hunk 1 above)
Comment (pderop) [reply example/src/test/java/com/example/BlockingDisallowTest.java]: applied feedback, thanks.

**Comment by pderop** [comment]
@chemicL , thanks for the review
@simonbasle , thanks for the proposed junit test

**Pull request merged**